scripts/analyze_deps.py

```diff
@@ -269,17 +283,25 @@ def render_report(output_path, report_context):
                         for lib in libs:
                             print("  * %s" % (lib))
         if len(changed_reqs) > 0:
-            exitcode = 1
-            if args.verbose:
-                for changed_req in changed_reqs:
-                    [frozen_spec] = frozen[changed_req]
-                    for current_spec in dependencies[changed_req]:
-                        if frozen_spec == current_spec:
-                            continue
-                        libs = dependencies[changed_req][current_spec]
-                        print("\nThe following libraries declare requirement '%s' which does not match the frozen requirement '%s':" % (changed_req + current_spec, changed_req + frozen_spec))
-                        for lib in libs:
-                            print("  * %s" % (lib))
+            for changed_req in changed_reqs:
+                frozen_specs, current_specs = changed_reqs[changed_req]
+                unmatched_specs = set(current_specs) - set(frozen_specs)
+                override_specs = overrides.get(changed_req, [])
+
+                for spec in unmatched_specs:
+                    non_overridden_libs = []
```

Comment:
Not sure if it's possible to rewrite this way, because not good at python, but a one line assignment to `non_overridden_libs` would improve clarity on what this code does, IMO.

Reply:
Also, does the new array need to be constructed here? Seems like it's re-assigned no matter what.

Reply:
I think this was an artifact from a previous approach I threw out, pretty sure you're right, I should just be able to remove L292 entirely. Thanks!

Reply:
Fixed in c757930c6